src/dependency/Pyegi/Pyegi/minimals/minimal_pkg_installer.py

```diff
@@ -0,0 +1,351 @@
+    _initialize_script_dir(script_path)
+    # infer python
+    py_version = infer_python_version(
+        script_path + GLOBAL_PATHS.pyproject_filename
+    ).python_version
+    com_dir = py_version.common_dir
+    # renew venvs
+    py_version.create_env(script_path)
+    py_version.create_env(GLOBAL_PATHS.temp_dir)
+
+    # start installing process
+    # create lock file
+    old_cwd = os.getcwd()
+    os.chdir(script_path)
+    result = py_version.run_module(["poetry", "lock"])
+    print(result.stdout)
+    # copy pyproject and lock file to temp
+    src = script_path + GLOBAL_PATHS.pyproject_filename
+    dst = GLOBAL_PATHS.temp_dir + GLOBAL_PATHS.pyproject_filename
+    shutil.copyfile(src, dst)
+    src = script_path + GLOBAL_PATHS.poetry_lock_filename
+    dst = GLOBAL_PATHS.temp_dir + GLOBAL_PATHS.poetry_lock_filename
+    shutil.copyfile(src, dst)
+    # analyse lock file
+    lock_content = toml.load(script_path + GLOBAL_PATHS.poetry_lock_filename)
+    packages = lock_content["package"]
+    new_packages = []
+    for package in packages:
+        if package["category"] != "main":
+            continue
+        path_compatible_pkg_name: str = package["name"]
+        path_compatible_pkg_name = path_compatible_pkg_name.replace("-", "_")
+        name_in_commons = f"{path_compatible_pkg_name}-{package['version']}"
+        # update lib_links
+        add_script_to_lib_links(script_id, name_in_commons, com_dir)
+        # check if the package is already in commons
+        dir_exist = os.path.isdir(com_dir + name_in_commons)
+        if dir_exist:
+            # create links for existing packages
+            commonize_pkg(
+                com_dir, name_in_commons, [GLOBAL_PATHS.temp_dir, script_path]
+            )
+        else:
+            new_packages.append(name_in_commons)
+    # install new packages
+    os.chdir(GLOBAL_PATHS.temp_dir)
+    result = py_version.run_module(["poetry", "install", "--no-dev", "--no-root"])
+    print(result.stdout)
+    # update common-packages and create links
+    for name_in_commons in new_packages:
+        try:
+            commonize_pkg(
+                com_dir,
+                name_in_commons,
+                [script_path],
+                src=GLOBAL_PATHS.temp_dir,
+                is_new=True,
+            )
+        except FileNotFoundError as e:
+            warnings.warn(e.args[0])
+    # revert current location
+    os.chdir(old_cwd)
+
+    # removing temp dir
+    shutil.rmtree(GLOBAL_PATHS.temp_dir)
+    # remove unused packages from commons
+    clean_lib_links(com_dir)
+
+    # update feed file
+    if is_feed:
+        from utils import FeedFile, InstallationStatus
+
+        feed_file = FeedFile()
+        script = feed_file.get_script(script_id)
+        script.installation_status = InstallationStatus.COMPLETED
+        feed_file.update_script(script)
+
+    print(f"Processed {script_id} dependencies.")
+    return py_version
```

Comment:
I wish there was a feature in Github which would link the new files to deleted files so that there was a comparison between those. I feel like some of these lines existed in the deleted files but don't know what has been modified.